pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3178,6 +3181,7 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
       String errorMsg = String.format("Failed to update the segment lineage during startReplaceSegments. "
           + "(tableName = %s, segmentsFrom = %s, segmentsTo = %s)", tableNameWithType, segmentsFrom, segmentsTo);
       LOGGER.error(errorMsg, e);
+      _controllerMetrics.addMeteredTableValue(tableNameWithType, ControllerMeter.NUMBER_START_REPLACE_FAILURE, 1);
```

Comment:
Can these metrics be emitted on resource level? By doing so, you don't have to inject the controllerMetrics class into the resource manager.

Reply:
Another place that we can emit the metrics is `PinotSegmentUploadDownloadRestletResource`. We already inject controllerMetrics object there.

Reply:
Good point, updated to emit in resource manager.